python/tvm/relax/block_builder.py

```diff
@@ -269,9 +270,7 @@ def function(
         ret: FunctionScope
             A FunctionScope for building a Relax function node.
         """
-        if not params:
-            params = None
```

Comment:
I think in case `param = None`, we should make it an empty list rather than keep it as `None`?

Reply:
Here when param = None, it means parameters will be updated in emit_func_output. So I guess the only case is param is an empty list or tuple?

Reply:
> Here when param = None, it means parameters will be updated in emit_func_output.

Could you elaborate a bit more? `emit_func_output` deals with the return value of a Relax function, while here `params` is about the input parameters.

Reply:
Okay! `emit_func_output` has an optional parameter called `params`. We could not specify function parameters in `bb.function()` but specify it in `emit_func_output`, so we can collect parameters simultaneously in the process of building functions.

Reply:
So when `params` here is None as the default value, it means that user doesn't specify parameters in `bb.function()`.

Reply:
Ah thanks for letting me know! I don’t quite know `emit_func_output` has such functionality before... Okay then your change makes sense and let’s go with it.